diff -urbB libdomainkeys-0.68.org/Makefile libdomainkeys-0.68/Makefile
--- libdomainkeys-0.68.org/Makefile	2005-08-30 22:31:39.000000000 +0200
+++ libdomainkeys-0.68/Makefile	2005-09-29 21:24:08.000000000 +0200
@@ -2,7 +2,7 @@
 CFLAGS=-DBIND_8_COMPAT -O2
 #CFLAGS += -DDK_DEBUG -DDK_HASH_BUFF -Wall
 INCS=-I.
-LIBS=-L. -ldomainkeys -lcrypto
+LIBS=-lcrypto -lresolv
 MAKE=make
 
 UNAME := $(shell uname)
@@ -14,49 +14,44 @@
 	CFLAGS += -DUNIXWARE
 endif
 
+dktest: dktest.lo libdomainkeys.la
+	libtool --mode=link --tag=CXX $(CC) $(INCS) $(CFLAGS) -o $@ dktest.lo libdomainkeys.la $(LIBS)
 
-dktest: dktest.o libdomainkeys.a dns.lib
-	$(CC) $(CFLAGS) -o dktest dktest.o $(LIBS) `cat dns.lib`
+dktest.lo: dktest.c domainkeys.h libdomainkeys.la
+	libtool --mode=compile --tag=CXX $(CC) $(INCS) $(CFLAGS) -o $@ -c $<
 
-dktest.o: dktest.c domainkeys.h
-	$(CC) $(CFLAGS) -c dktest.c $(INCS)
-
-testtrace: testtrace.o dktrace.o dktrace.h
-	$(CC) $(CFLAGS) -o testtrace testtrace.o dktrace.o
-
-dktrace.o: dktrace.c dktrace.h
+testtrace: testtrace.lo dktrace.lo dktrace.h libdomainkeys.la
+	libtool --mode=link --tag=CXX $(CC) $(INCS) $(CFLAGS) -o $@ $< $(LIBS)
 
 makeheader: makeheader.c
-	$(CC) $(CFLAGS) -o makeheader makeheader.c
+	libtool --mode=link --tag=CXX $(CC) $(INCS) $(CFLAGS) -o $@ $<
+
 domainkeys.o: domainkeys.c
-	$(CC) $(CFLAGS) -c domainkeys.c $(INCS)
-dns_txt.o: dns_txt.c
-	$(CC) $(CFLAGS) -c dns_txt.c
-dnstest.o: dnstest.c
-	$(CC) $(CFLAGS) -c dnstest.c
-dnstest: dnstest.o dns_txt.o libdomainkeys.a dns.lib
-	$(CC) $(CFLAGS) -o dnstest dnstest.o dns_txt.o `cat dns.lib` $(LIBS) 
-dns.lib: dnstest.c
-	(if $(MAKE) dnstest >/dev/null 2>&1; then echo -lresolv; else echo ""; fi) >dns.lib
+	libtool --mode=compile --tag=CXX $(CC) $(INCS) $(CFLAGS) -o $@ -c $<
+	
+dnstest.lo: dnstest.c
+	libtool --mode=compile --tag=CXX $(CC) $(INCS) $(CFLAGS) -o $@ -c $<
+	
+dnstest: dnstest.lo dns_txt.lo libdomainkeys.la
+	libtool --mode=link --tag=CXX $(CC) $(INCS) $(CFLAGS) -o $@ $< $(LIBS)
+	
 domainkeys.h: domainkeys.c makeheader
 	./makeheader <domainkeys.c >domainkeys.h
 
-libdomainkeys.a: domainkeys.o dns_txt.o dktrace.o
-	rm -f libdomainkeys.a
-	ar cr libdomainkeys.a domainkeys.o dns_txt.o dktrace.o
-	ranlib libdomainkeys.a
-
-python: domainkeys_wrap.o _domainkeys.so
-
-domainkeys_wrap.o: domainkeys.h domainkeys.i domainkeys_wrap.c
-	#swig -python -module domainkeys domainkeys.i
-	$(CC) $(CFLAGS) -c domainkeys_wrap.c  -I/usr/include/python2.4/
+domainkeys.lo dns_txt.lo dktrace.lo: %.lo: %.c
+	libtool --mode=compile --tag=CXX $(CC) $(INCS) $(CFLAGS) $(OPTS) -o $@ -c $<
+	
+libdomainkeys.la: domainkeys.lo dns_txt.lo dktrace.lo
+	libtool --mode=link --tag=CXX $(CC) domainkeys.lo dns_txt.lo dktrace.lo -o $@ -rpath /usr/lib -version-info 0:66 $(LIBS)
+
+domainkeys_wrap.o: domainkeys_wrap.c
+	$(CC) $(CFLAGS) -c domainkeys_wrap.c  -I /usr/include/python2.2/
 
-_domainkeys.so: domainkeys.o dns_txt.o dktrace.o domainkeys_wrap.o
-	ld -shared domainkeys.o dns_txt.o dktrace.o domainkeys_wrap.o -lcrypto -lresolv -o _domainkeys.so
+_domainkeys.so: domainkeys.o dns_txt.o
+	ld -shared domainkeys.o dns_txt.o domainkeys_wrap.o -lcrypto -lresolv -o _domainkeys.so
 
 clean:
-	rm -f *.o *.so libdomainkeys.a dns.lib dnstest makeheader dktest testtrace domainkeys.h
+	rm -f *.o libdomainkeys.a dns.lib dnstest makeheader dktest testtrace domainkeys.h
 
 # 
 distributionfile:
